plugins/modules/sssd_info.py

```diff
@@ -0,0 +1,300 @@
+
+    def check_domain_status(self, domain: str) -> str:
+        """Check if domain is online.
+
+        Args:
+            domain: Domain name to check.
+
+        Returns:
+            'online' if domain is online, 'offline' otherwise.
+        """
+        domain_obj = self._get_domain_object(domain)
+        iface = dbus.Interface(domain_obj, dbus_interface=self.DOMAIN_INTERFACE)
+        return 'online' if iface.IsOnline() else 'offline'
+
+    def get_active_servers(self, domain: str, server_type: str) -> dict:
+        """Get active servers for domain.
+
+        Args:
+            domain: Domain name to get servers for.
+            server_type: Type of servers ('IPA' or 'AD').
+
+        Returns:
+            Dictionary with server types as keys and server names as values.
+        """
+        domain_obj = self._get_domain_object(domain)
+        iface = dbus.Interface(domain_obj, dbus_interface=self.DOMAIN_INTERFACE)
+
+        if server_type == 'IPA':
+            server_name = f'{server_type} Server'
+            return {server_name: iface.ActiveServer(server_type)}
+        elif server_type == 'AD':
+            return {
+                'AD Global Catalog': iface.ActiveServer(f'sd_gc_{domain}'),
+                'AD Domain Controller': iface.ActiveServer(f'sd_{domain}')
+            }
+        return {}
+
+    def list_servers(self, domain: str, server_type: str) -> list:
+        """List all servers for domain.
+
+        Args:
+            domain: Domain name to list servers for.
+            server_type: Type of servers ('IPA' or 'AD').
+
+        Returns:
+            List of server names.
+        """
+        domain_obj = self._get_domain_object(domain)
+        iface = dbus.Interface(domain_obj, dbus_interface=self.DOMAIN_INTERFACE)
+        if server_type == 'IPA':
+            return iface.ListServers(server_type)
+        elif server_type == 'AD':
+            return iface.ListServers(f'sd_{domain}')
+        return []
+
+    def get_domain_list(self) -> list:
+        """Get list of all domains.
+
+        Returns:
+            List of domain names.
+        """
+        response = self.infopipe_iface.ListDomains()
+        return [str(domain).rsplit('/', maxsplit=1)[-1].replace('_2e', '.') for domain in response]
+
+
+def main() -> None:
+    """Main function for the Ansible module."""
+    module = AnsibleModule(
+        argument_spec={
+            'action': {
+                'type': 'str',
+                'required': True,
+                'choices': ['domain_status', 'domain_list', 'active_servers', 'list_servers']
+            },
+            'domain': {'type': 'str'},
+            'server_type': {'type': 'str', 'choices': ['IPA', 'AD']}
+        },
+        supports_check_mode=True,
+        required_if=[
+            ('action', 'domain_status', ['domain']),
+            ('action', 'list_servers', ['domain', 'server_type']),
+            ('action', 'active_servers', ['domain', 'server_type'])
+        ]
```

Comment:
Not a mandatory thing, but for consistency most (if not all) of the modules declare this using `dict()` instead of the literal `{}`. There's a debate to be had about performance of one versus the other, but I don't really think that is much relevant in the context of Ansible modules.

Also, please add the comma after the last element of the list or dictionary. It reduces clutter in the PR when the modules get additions in the future.